File: Documentation/5.4/Samples/python/ClientApi/Session/Querying/TextSearch/ExactMatch.py
from ravendb import WhereParams

from examples_base import ExampleBase, Employee, Order


class ExactMatch(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_example(self):
        with self.embedded_server.get_document_store("ExactMatch") as store:
            with store.open_session() as session:
                # region exact_1
                employees = list(
                    session.
                    # Make a query on 'Employees' collection
                    query(object_type=Employee)
                    # Query for all documents where 'FirstName' equals 'Robert'
                    # Pass 'exact=True' for a case-sensitive match
                    .where_equals("FirstName", "Robert", exact=True)
                )
                # endregion

            with store.open_session() as session:
                # region exact_4
                orders = list(
                    session
                    # Make a query on 'Orders' collection
                    .query(object_type=Order)
                    # Query for documents that contain at least one order line with 'Teatime Chocolate Biscuits'
                    .where_equals(
                        "Lines[].ProductName",
                        "Teatime Chocolate Biscuits",
                        # Pass 'exact=True' for a case-sensitive match
                        exact=True,
                    )
                )
                # endregion

File: Documentation/5.4/Samples/php/ClientApi/session/Querying/TextSearch/ExactMatch.php
<?php

use RavenDB\Documents\Session\DocumentQueryInterface;
use RavenDB\Samples\Infrastructure\DocumentStoreHolder;

interface FooInterface
{
    # region syntax
    function whereEquals(string $fieldName, $value, bool $exact = false): DocumentQueryInterface;
    function whereNotEquals(string $fieldName, $value, bool $exact = false): DocumentQueryInterface;

    // ... rest of where methods
    # endregion
}

class ExactMatch
{
    public function examples(): void
    {
        $store = DocumentStoreHolder::getStore();
        try {
            $session = $store->openSession();
            try {
                # region exact_1
                // load all entities from 'Employees' collection
                // where FirstName field's contents equals 'Robert' (case sensitive match)

                /** @var array<Employee> $employees */
                $employees = $session->query(Employee::class)
                    ->whereEquals("FirstName", "Robert", true)
                    ->toList();
                # endregion
            } finally {
                $session->close();
            }

            $session = $store->openSession();
            try {
                # region exact_4
                // return all entities from the 'Orders' collection which contain 
                // at least one order line with 'Teatime Chocolate Biscuits' product
                // perform a case-sensitive match
                /** @var array<Order> $orders */
                $orders = $session->query(Order::class)
                    ->whereEquals("Lines[].ProductName", "Teatime Chocolate Biscuits", true)
                    ->toList();
                # endregion
            } finally {
                $session->close();
            }
        } finally {
            $store->close();
        }
    }
}

class Employee
{
}

class Order
{
}
